bson/decoder_test.go

```diff
@@ -30,10 +31,7 @@ func TestBasicDecode(t *testing.T) {
 			noerr(t, err)
 			err = decoder.DecodeValue(bsoncodec.DecodeContext{Registry: reg}, vr, got)
 			noerr(t, err)
-
-			if !reflect.DeepEqual(got.Addr().Interface(), tc.want) {
-				t.Errorf("Results do not match. got %+v; want %+v", got, tc.want)
-			}
+			assert.Equal(t, tc.want, got.Addr().Interface(), "Results do not match.")
```

Comment:
Why move away from `reflect.DeepEqual` here? Does `testify/assert` do the correct recursive comparison?

Reply:
The `reflect.DeepEqual` comparison is correct, but the information that is printed on failure is not helpful. For example, here's the previous printed test failure message:
```
Results do not match. got {I:0xc0000ad698 M:0xc0000aa140 B:0xc0000be8b8 S:0xc0000a19a0}; want &{I:0xc0000ad140 M:0xc0000aa098 B:0xc0000be288 S:0xc0000a12a0}
```
And the new one:
```
Error Trace:	decoder_test.go:34
Error:      	Not equal: 
                expected: &bson.myStruct{I:(*bson.myInt64)(0xc00011f140), M:(*bson.myMap)(0xc00011a098), B:(*bson.myBytes)(0xc000132288), S:(*bson.myString)(0xc0001112a0)}
                actual  : &bson.myStruct{I:(*bson.myInt64)(0xc00011f9f8), M:(*bson.myMap)(0xc00011a140), B:(*bson.myBytes)(0xc000132948), S:(*bson.myString)(0xc000111a70)}
                
                Diff:
                --- Expected
                +++ Actual
                @@ -2,6 +2,4 @@
                    I: (*bson.myInt64)(0),
                - M: (*bson.myMap)({
                - }),
                - B: (*bson.myBytes)({
                - }),
                + M: (*bson.myMap)(<nil>),
                + B: (*bson.myBytes)(<nil>),
                    S: (*bson.myString)("")
Test:       	TestBasicDecode/TODO
Messages:   	Results do not match.
```